### ISSUE TITLE
Menu: wrong calculated position with overlay="true"

### ISSUE THREAD

**Issue by ReinhardAT** [primefaces/primefaces#6906]
Same defect as #4536. Defect was solved in 7.x but comes back with 8.0

## 1) Environment
PrimeFaces version: 8.0.4
Application server + version: TomEE 8.0.4
Affected browsers: All

## 2) Expected behavior
Calculated position of tag p:menu (overlay="true") on p:dialog is not dependent on the position of the scroll bar main window.

## 3) Actual behavior
Calculated position of tag p:menu (overlay="true") on p:dialog is dependent on the position of the scroll bar main window.

## 4) Steps to reproduce
Create page with vertival scroll bar. Open p:dialog with button with p:menu (overlay="true").
p:menu position is dependent on the location of the scroll bar.

## 5) Sample XHTML
``` 
<html xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://java.sun.com/jsf/html"
	xmlns:f="http://java.sun.com/jsf/core"
	xmlns:p="http://primefaces.org/ui">

<h:head>
</h:head>

<h:body style="height: 1500px;">
	<p:commandButton value="Modal" type="button" onclick="PF('dlg2').show();" />

	<p:dialog header="Modal Dialog" widgetVar="dlg2" modal="true" position="top" height="150">
		<h:form>
			<h:outputText value="This is a Modal Dialog." />
			<h3>Overlay</h3>
			<p:commandButton id="dynaButton" value="Show menu" type="button" />
			<p:menu overlay="true" trigger="dynaButton" my="left top" at="left bottom">
				<p:menuitem value="Save" />
				<p:menuitem value="Update" />
			</p:menu>
		</h:form>
	</p:dialog>
</h:body>
</html>
```

**Comment by melloware** [comment]
Reproducer: 
[pf-6906.zip](https://github.com/primefaces/primefaces/files/5888462/pf-6906.zip)

Confirmed happening in 8.0 and 10.0.0.RC1

**Comment by melloware** [comment]
it looks like it was caused by this issue: https://github.com/primefaces/primefaces/issues/5035

**Comment by melloware** [comment]
If you need this fix right now just add this code to you app:
```javascript
if (PrimeFaces.widget.Menu) {
    PrimeFaces.widget.Menu.prototype.show = function() {
        this.jq.css({
            'z-index': ++PrimeFaces.zindex,
            'visibility': 'hidden'
        }).show();

        this.align();
        this.jq.css('visibility', '');
    };
}
```

**Comment by ReinhardAT** [comment]
Thanks a lot for the quick fix. It works for me.

**Comment by melloware** [comment]
This is fixed in 10.0.0-RC2 with this fix: #6973
